pkg/asset/machines/master.go

```diff
@@ -615,6 +616,24 @@ func (m *Master) Generate(ctx context.Context, dependencies asset.Parents) error
 		machineConfigs = append(machineConfigs, ignIPv6)
 	}
 
+	if installConfig.Config.EnabledFeatureGates().Enabled(features.FeatureGateMultiDiskSetup) {
+		for i, diskSetup := range installConfig.Config.ControlPlane.DiskSetup {
+			var dataDisk any
+			switch ic.Platform.Name() {
+			case azuretypes.Name:
+				azureControlPlaneMachinePool := ic.ControlPlane.Platform.Azure
+				dataDisk = azureControlPlaneMachinePool.DataDisks[i]
+			case vspheretypes.Name:
+				// todo, putting it here to get the linter to stop bugging me
```

Comment:
would it make sense to just bypass the problematic rule in this case rather than a dead block?